src/App.tsx

```diff
@@ -1,61 +1,120 @@
+import { useState } from 'react';
 import './App.scss';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todosWithUserProp: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, updateTodos] = useState(todosWithUserProp);
+  const [user, setUser] = useState(0);
+  const [title, setTitle] = useState('');
+  const [hasUser, setHasUser] = useState(true);
+  const [hasTitle, setHasTitle] = useState(true);
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        id="form"
+        onSubmit={(event) => {
+          event.preventDefault();
```

Comment:
Move the onSubmit logic to a separate function. It follows the best practice of separating concerns and keeping your JSX code clean and focused on rendering components rather than defining complex logic directly within it.